hoomd/ForceCompute.cc

```diff
@@ -377,23 +397,38 @@ pybind11::object ForceCompute::getForcesPython()
         dims[0] = 0;
         dims[1] = 0;
         }
-    std::vector<vec3<double>> force(dims[0]);
+    std::vector<vec3<double>> global_force(dims[0]);
 
-    // This is slow: TODO implement a proper gather operation
-    for (unsigned int i = 0; i < m_pdata->getNGlobal(); i++)
+    // sort forces by particle tag
+    std::vector<vec3<double>> local_force(m_pdata->getN());
+    std::vector<uint32_t> local_tag(m_pdata->getN());
+    ArrayHandle<unsigned int> h_tag(m_pdata->getTags(), access_location::host, access_mode::read);
+    ArrayHandle<unsigned int> h_rtag(m_pdata->getRTags(), access_location::host, access_mode::read);
+    ArrayHandle<Scalar4> h_force(m_force, access_location::host, access_mode::read);
+    std::copy(h_tag.data, h_tag.data + m_pdata->getN(), local_tag.begin());
+    std::sort(local_tag.begin(), local_tag.end());
+    for (unsigned int i = 0; i < m_pdata->getN(); i++)
         {
-        Scalar3 f = getForce(i);
-        if (root)
-            {
-            force[i].x = f.x;
-            force[i].y = f.y;
-            force[i].z = f.z;
-            }
+        local_force[i].x = h_force.data[h_rtag.data[local_tag[i]]].x;
+        local_force[i].y = h_force.data[h_rtag.data[local_tag[i]]].y;
+        local_force[i].z = h_force.data[h_rtag.data[local_tag[i]]].z;
+        }
+
+    if (m_sysdef->isDomainDecomposed())
+        {
+#ifdef ENABLE_MPI
+        m_gather_tag_order.setLocalTagsSorted(local_tag);
+        m_gather_tag_order.gatherArray(global_force, local_force);
+#endif
+        }
+    else
+        {
+        global_force = local_force;
```

Comment:
```suggestion
        global_force = std::move(local_force);
```
See previous comment about `std::move`.